pre_commit/languages/dotnet.py

```diff
@@ -0,0 +1,108 @@
+import contextlib
+import glob
+import os.path
+import shutil
+from typing import Generator
+from typing import Sequence
+from typing import Tuple
+
+import pre_commit.constants as C
+from pre_commit.envcontext import envcontext
+from pre_commit.envcontext import PatchesT
+from pre_commit.envcontext import Var
+from pre_commit.hook import Hook
+from pre_commit.languages import helpers
+from pre_commit.prefix import Prefix
+from pre_commit.util import clean_path_on_failure
+from pre_commit.util import rmtree
+
+ENVIRONMENT_DIR = 'dotnetenv'
+BIN_DIR = 'bin'
+BUILD_CONFIG = 'Release'
+
+get_default_version = helpers.basic_get_default_version
+healthy = helpers.basic_healthy
+
+
+def get_env_patch(venv: str) -> PatchesT:
+
+    # https://github.com/dotnet/sdk/issues/9304
+    dotnet = shutil.which('dotnet')
+    assert dotnet, 'No dotnet executable found - please install one'
```

Comment:
if I'm reading this correctly, we shouldn't need to do this as the host machine should have that set up properly already -- I think all that needs to happen here is a `passenv` in `tox.ini`?

(seems silly that they didn't just default it to the location where `argv[0]` is, but I didn't have time to read the whole dotnet issue!)

Reply:
You're right; I think this should be fine to remove. I added it initially since I have a non-standard install and ran into that issue. I've now got `DOTNET_ROOT` set in my env though and the hook still seems to work without this patch.

> (seems silly that they didn't just default it to the location where argv[0] is, but I didn't have time to read the whole dotnet issue!)

Welcome to the wonderful world of c# tooling 😊